agent/utils/sign_http_request.go

```diff
@@ -20,10 +20,16 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/signer/v4"
+	"github.com/cihub/seelog"
 )
 
 // SignHTTPRequest signs an http.Request struct with authv4 using the given region, service, and credentials.
-func SignHTTPRequest(req *http.Request, region, service string, creds *credentials.Credentials, body io.ReadSeeker) {
+func SignHTTPRequest(req *http.Request, region, service string, creds *credentials.Credentials, body io.ReadSeeker) error {
 	signer := v4.NewSigner(creds)
-	signer.Sign(req, body, service, region, time.Now())
+	_, err := signer.Sign(req, body, service, region, time.Now())
```

Comment:
what does the signer return?

Reply:
it always returned the [header](https://github.com/aws/aws-sdk-go/blob/b2e866cc3d25372c6bde1ce18dbbf11862e954e9/aws/signer/v4/v4.go#L271)

Reply:
Why don't we need that?

Reply:
looks like we pass in a pointer to the request and the signer does the `Request.Header.Set` things